clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -79,6 +84,46 @@ public static boolean hasId(int id) {
         return id >= MIN_API_KEY && id <= MAX_API_KEY;
     }
 
+    public short latestVersion() {
+        if (id >= Protocol.CURR_VERSION.length)
```

Comment:
Hmm... seems unnecessary since you'd have to explicitly add a new api with a negative id. If we put the check anyway, maybe it should be in the `ApiKeys` constructor? I'm also tempted to call this check unnecessary, but it's a bit harder to justify removing since the array lives in a separate class.